packages/bot-web-ui/src/components/dashboard/dbot-tours/mobile-tours/__tests__/bot-builder-tour.spec.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { mockStore, StoreProvider } from '@deriv/stores';
+import { act, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { mock_ws } from 'Utils/mock';
+import RootStore from 'Stores/root-store';
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import BotBuilderTour from '../bot-builder-tour';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+jest.mock('@deriv/deriv-charts', () => ({
+    setSmartChartsPublicPath: jest.fn(),
+}));
+
+describe('<BotBuilderTour />', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+
+    beforeAll(() => {
+        const mock_store = mockStore({});
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    it('renders botbuilder tour', () => {
+        const { container } = render(<BotBuilderTour />, {
+            wrapper,
+        });
+        expect(container).toBeInTheDocument();
+    });
+
+    it('shows the tour content when tour starts', () => {
+        render(<BotBuilderTour />, {
+            wrapper,
+        });
+        const tourElement = screen.getByTestId('botbuilder-tour-mobile');
+        expect(tourElement).toHaveClass('dbot-slider__bot-builder-tour');
+    });
+
+    it('Check if next button is clicked', async () => {
+        render(<BotBuilderTour />, {
+            wrapper,
+        });
+        const nextButton = screen.getByTestId('next-bot-builder-tour');
+        if (!nextButton) {
+            console.error('Next button not found. Make sure it exists in your component.'); // eslint-disable-line no-console
+        }
+        const onClickMock = jest.fn();
+        nextButton.onclick = onClickMock;
+        act(() => {
+            userEvent.click(nextButton);
+        });
+        await waitFor(() => {
+            expect(onClickMock).toHaveBeenCalledTimes(1);
```

Comment:
Can we test the side effects of this button click? Whether the right action is being performed or not? Instead of testing whether the button is getting clicked or not.